numpy/lib/nanfunctions.py

```diff
@@ -991,7 +995,10 @@ def _nanpercentile1d(arr1d, q, overwrite_input=False, interpolation='linear'):
     s = np.where(c)[0]
     if s.size == arr1d.size:
         warnings.warn("All-NaN slice encountered", RuntimeWarning)
-        return np.nan
+        if q.ndim == 0:
+            return np.nan
+        else:
+            return np.nan * np.ones((len(q),))
```

Comment:
I didn't consider that because it doesn't appear that percentile supports the higher dimensional case either.  The following results in a ValueError being raised.

```python
import numpy as np
data = np.eye(5, dtype=np.float64)
np.percentile(data, [[60, 80], [90, 100]], keepdims=True, axis=1)
```